bigtable/client/row_reader.cc

```diff
@@ -156,8 +162,12 @@ void RowReader::Advance(internal::OptionalRow& row) {
     }
 
     if (not status.ok() and not retry_policy_->on_failure(status)) {
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
       internal::RaiseRuntimeError("Unretriable error: " +
                                   status.error_message());
+#else
+      return;
```

Comment:
This means that the `RowReader` returned by `noex::Table::ReadRows` may raise exceptions.

Is that what we want?  You have not answered my question regarding whether this is an intermediate change for RowReader and you plan a future one to fix #247.  As an intermediate fix it is a bit funky, but understandable.

Reply:
Also, when compiled without exceptions this means that the errors are ignored if the user forgets to call `finish()` and check the returned `grpc::Status`.